File: julia/work/2012/1B/Ctcgen.py
import random
random.seed(8675309)
(fn,ntc) = ("Ctc1.in",1000)
with open(fn,'wt') as fp :
    print(ntc, file=fp)
    for i in range(ntc) :
        N = 500
        s = set()
        while len(s) < 500 :
            b = random.randrange(1,100000+1)
            s.add(b)
        a = [x for x in s]
        random.shuffle(a)
        a = [500] + a
        astr = " ".join(str(x) for x in a)
        print(astr,file=fp)

File: work/2014/1A/Ctcgen.jl
using Random
Random.seed!(8675309)

function goodperm()
    a = [x for x in 0:999]
    for i in 1:1000
        p = Random.rand(i:1000)
        a[i],a[p] = a[p],a[i]
    end
    return a
end

function badperm()
    a = [x for x in 0:999]
    for i in 1:1000
        p = Random.rand(1:1000)
        a[i],a[p] = a[p],a[i]
    end
    return a
end

print("120\n")
for i in 1:120
    print("1000\n")
    x = Random.rand()
    p = x < 0.5 ? goodperm() : badperm()
    pstr = join(p, " ")
    print("$pstr\n")
end